components/collector/src/base_collectors/collector.py

```diff
@@ -43,41 +40,37 @@ def record_health() -> None:
 
     async def start(self) -> NoReturn:
         """Start fetching measurements indefinitely."""
-        timeout = aiohttp.ClientTimeout(total=120)
-        while True:
-            self.record_health()
-            # The TCPConnector has limit 0, meaning unlimited, because aiohttp only closes connections when the response
-            # is closed. But due to the architecture of the collector (first collect all the responses, then parse them)
-            # the responses are kept around relatively long, and hence the connections too. To prevent time-outs while
-            # aiohttp waits for connections to become available we don't limit the connection pool size.
-            async with aiohttp.ClientSession(
-                connector=aiohttp.TCPConnector(limit=0, ssl=False),
-                raise_for_status=True,
-                timeout=timeout,
-                trust_env=True,
-            ) as session:
-                with timer() as collection_timer:
-                    await self.collect_metrics(session)
-            sleep_duration = max(0, config.MAX_SLEEP_DURATION - collection_timer.duration)
-            logging.info(
-                "Collecting took %.1f seconds. Sleeping %.1f seconds...",
-                collection_timer.duration,
-                sleep_duration,
-            )
-            await asyncio.sleep(sleep_duration)
+        # The TCPConnector has limit 0, meaning unlimited, because aiohttp only closes connections when the response
+        # is closed. But due to the architecture of the collector (first collect all the responses, then parse them)
+        # the responses are kept around relatively long, and hence the connections too. To prevent time-outs while
+        # aiohttp waits for connections to become available we don't limit the connection pool size.
+        async with aiohttp.ClientSession(
+            connector=aiohttp.TCPConnector(limit=0, ssl=False),
+            raise_for_status=True,
+            timeout=aiohttp.ClientTimeout(total=120),
```

Comment:
Shouldn't this 120 be defined as a constant (also for usage in tests), or even an env var?

Reply:
I'll make it an environment variable.